### PR TITLE
Add hl.utils.range_matrixtable.

### PR DESCRIPTION
**Pull request by tpoterba**
Move Table.range to hl.utils.range_table

### CODE REVIEW

File: python/hail/docs/utils/index.rst [hunk 1]
```diff
@@ -11,4 +11,8 @@ utils
 
 .. autofunction:: hadoop_write
 
+.. autofunction:: range_table
+
+.. autofunction:: range_matrixtable
```
Comment (jbloom22) [inline python/hail/docs/utils/index.rst]: to be consistent with read and import, this should be `range_matrix_table`

File: python/hail/utils/misc.py [hunk 2]
```diff
@@ -1,10 +1,89 @@
...
-from hail.typecheck import enumeration
+from hail.typecheck import enumeration, typecheck, nullable
 import difflib
 from collections import defaultdict
 import os
 
 
+@typecheck(n_rows=int, n_cols=int, n_partitions=nullable(int))
+def range_matrixtable(n_rows, n_cols, n_partitions=None):
+    """Construct a dummy matrix table with `n_rows` rows and `n_cols` columns.
```
Comment (jbloom22) [inline python/hail/utils/misc.py]: Construct a matrix table with row and column indices and no entry fields.

File: python/hail/utils/misc.py [hunk 3]
```diff
@@ -1,10 +1,89 @@
...
 from hail.utils.java import handle_py4j, Env, joption, error
-from hail.typecheck import enumeration
+from hail.typecheck import enumeration, typecheck, nullable
 import difflib
 from collections import defaultdict
 import os
 
 
+@typecheck(n_rows=int, n_cols=int, n_partitions=nullable(int))
+def range_matrixtable(n_rows, n_cols, n_partitions=None):
```
Comment (cseed) [inline python/hail/utils/misc.py]: `range_matrix_table` to match `read_matrix_table`?

File: python/hail/utils/misc.py [hunk 4]
```diff
@@ -1,10 +1,89 @@
...
+
+        >>> ds.count_rows()
+        100
+
+        >>> ds.count_cols()
+        10
+
+    Notes
+    -----
+    The matrix table will contain the following fields:
```
Comment (jbloom22) [inline python/hail/utils/misc.py]: "The resulting matrix table contains the following fields"

File: python/hail/utils/misc.py [hunk 5]
```diff
@@ -1,10 +1,89 @@
...
+     - `col_idx` (:class:`.TInt32`) - Column index (column key).
+
+    It contains no entry fields.
+
+    This method is meant for testing and learning, and is not optimized for
+    production performance.
+
+    Parameters
+    ----------
+    n_rows : int
```
Comment (jbloom22) [inline python/hail/utils/misc.py]: I think we've been using
```
:obj:`int`
```
for parameters

File: python/hail/utils/misc.py [hunk 6]
```diff
@@ -1,10 +1,89 @@
...
+
+    Returns
+    -------
+    :class:`.MatrixTable`
+    """
+    return hl.MatrixTable(Env.hail().variant.MatrixTable.range(Env.hc()._jhc, n_rows, n_cols, joption(n_partitions)))
+
+@typecheck(n=int, n_partitions=nullable(int))
+def range_table(n, n_partitions=None):
+    """Construct a dummy table with `n`.
```
Comment (jbloom22) [inline python/hail/utils/misc.py]: Construct a table with row indices and no other fields.

File: python/hail/utils/misc.py [hunk 7]
```diff
@@ -1,10 +1,89 @@
...
+    .. doctest::
+
+        >>> df = hl.utils.range_table(100)
+
+        >>> df.count()
+        100
+
+    Notes
+    -----
+    The table will contain the following field:
```
Comment (jbloom22) [inline python/hail/utils/misc.py]: The resulting table contains one field:

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 8]
```diff
@@ -168,6 +168,48 @@ object MatrixTable {
...
+    val rdd = hc.sc.parallelize(0 until nRows, nPartitions.getOrElse(hc.sc.defaultMinPartitions))
+      .mapPartitions { it =>
+        val rv = RegionValue(Region())
+        val rvb = new RegionValueBuilder()
+        it.map { idx =>
+          rvb.set(rv.region)
+          rvb.start(localRVType)
+          rvb.startStruct()
+
+          // idx field
```
Comment (jbloom22) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: `// row_idx field`

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 9]
```diff
@@ -168,6 +168,48 @@ object MatrixTable {
...
+          rvb.addInt(idx)
+
+          // entries field
```
Comment (jbloom22) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: just ``// entries`` as there are no entries fields.

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 10]
```diff
@@ -168,6 +168,48 @@ object MatrixTable {
...
+          }
+          rvb.endArray()
+
+          rvb.endStruct()
+          rv.set(rv.region, rvb.end())
+          rv
+        }
+      }
+    val rvd = OrderedRVD(mt.orvdType, rdd, None, None)
+    new MatrixTable(hc, mt, Annotation.empty, (0 until nCols).map(Annotation(_)), rvd)
```
Comment (jbloom22) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: Prefer `Array.tabulate(nCols)(Annotation)` (or maybe `Annotation(_)`) to definitely get an Array

File: src/main/scala/is/hail/variant/MatrixTable.scala (hunk 9 above)
Comment (tpoterba) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: the array is a field of the top level struct.

File: python/hail/utils/misc.py [hunk 11]
```diff
@@ -1,10 +1,89 @@
...
+
+    Returns
+    -------
+    :class:`.MatrixTable`
+    """
+    return hl.MatrixTable(Env.hail().variant.MatrixTable.range(Env.hc()._jhc, n_rows, n_cols, joption(n_partitions)))
+
+@typecheck(n=int, n_partitions=nullable(int))
+def range_table(n, n_partitions=None):
+    """Construct a table row index and no other fields..
```
Comment (jbloom22) [inline python/hail/utils/misc.py]: "Construct a table with row indices and no other fields."

(you also have two periods at end)

**Pull request merged**